src/main/io/vtx.c

```diff
@@ -65,6 +71,36 @@ void vtxInit(void)
 {
 }
 
+static uint8_t getPowerAccordingToDistance(uint32_t distance){
+    /*
+        Range for power level:
+        power   min             max
+        1       0               distance_one
+        2       distance_one    distance_two
+        3       distance_two    distance_three
+        4       distance_three  distance_four
+        5       distance_four
+    */
+
+    if(distance < vtxSettingsConfig()->distance_one){
+        return 1;
+    }
+    else if(distance > vtxSettingsConfig()->distance_one && distance < vtxSettingsConfig()->distance_two){
+        return 2;
+    }
+    else if(distance > vtxSettingsConfig()->distance_two && distance < vtxSettingsConfig()->distance_three){
+        return 3;
+    }
+    else if(distance > vtxSettingsConfig()->distance_three 
+        && (vtxSettingsConfig()->distance_four == 0 || distance < vtxSettingsConfig()->distance_four)){
```

Comment:
Some VTXes only support 3 power levels (1,2,3), the code shouldn't rely on hard-coded constraints. Check run-time VTX capabilities instead.